pkg/zonemgr/zone_file.go

```diff
@@ -2,180 +2,151 @@ package zonemgr
-		zoneFile.userNameserverName, zoneFile.userAdminEmail)
-
-	//fmt.Println("======= prefix:\n", zoneFile.headerPref)
-
+	zoneFile.headerPref = fmt.Sprintf("$ORIGIN %s. \n$TTL %s \n@ IN SOA %s. %s. (", zoneFile.domain, ttl,
+		zoneFile.nameServerName, zoneFile.adminEmail)
 }
 
-/*
-									7200       ; refresh (2 hours)
-	                                3600       ; retry (1 hour)
-	                                1209600    ; expire (2 weeks)
-	                                3600       ; minimum (1 hour)
-	                                )
-	        IN      NS  ns1.mycluster.secondary.mydomain.com
-	        IN      A   1.2.3.4
-*/
 func (zoneFile *ZoneFile) generateHeaderSuffix() {
 	zoneFile.headerSuf = fmt.Sprintf(" %s %s %s %s)\n", refresh, retry, expire, ttl)
 
-	if zoneFile.userNameserverName != "" {
-		zoneFile.headerSuf += fmt.Sprintf("IN NS %s\n", zoneFile.userNameserverName)
+	if zoneFile.nameServerName != "" {
+		zoneFile.headerSuf += fmt.Sprintf("IN NS %s.\n", zoneFile.nameServerName)
 	}
 
-	if zoneFile.userNameserverIP != "" {
-		zoneFile.headerSuf += fmt.Sprintf("IN A %s\n", zoneFile.userNameserverIP)
+	if zoneFile.nameServerIP != "" {
+		zoneFile.headerSuf += fmt.Sprintf("IN A %s\n", zoneFile.nameServerIP)
 	}
 }
 
 func (zoneFile *ZoneFile) generateHeader() (header string) {
 	return zoneFile.headerPref + strconv.Itoa(zoneFile.soaSerial) + zoneFile.headerSuf
 }
 
-//////////// maps impl
-
-func (zoneFile *ZoneFile) updateVMIRecords(vmiName string, vmiNamespace string, interfaces []v1.VirtualMachineInstanceNetworkInterface) (isUpdated bool, err error) {
+func (zoneFile *ZoneFile) updateVMIRecords(vmiName string, vmiNamespace string, interfaces []v1.VirtualMachineInstanceNetworkInterface) (isUpdated bool) {
 	key := fmt.Sprintf("%s_%s", vmiName, vmiNamespace)
 	isUpdated = false
 	//delete vmi records from the list
 	if interfaces == nil {
-		delete(zoneFile.vmiRecordsMap, key)
-		isUpdated = true
-	} else {
-		ifacesMap := zoneFile.vmiRecordsMap[key]
-		//if not exist or changed - override
-		if ifacesMap == nil || isIfaceListChanged(ifacesMap, interfaces) {
-			zoneFile.vmiRecordsMap[key] = buildInterfacesMap(vmiName, vmiNamespace, interfaces)
+		if zoneFile.vmiRecordsMap[key] != nil {
+			delete(zoneFile.vmiRecordsMap, key)
 			isUpdated = true
 		}
+	} else {
+		newRecords := buildARecordsArr(vmiName, vmiNamespace, interfaces)
+		isUpdated = !isEqual(newRecords, zoneFile.vmiRecordsMap[key])
+		if isUpdated {
+			zoneFile.vmiRecordsMap[key] = newRecords
+		}
 	}
 
-	fmt.Println("********** zoneFile.vmiRecordsMap: ", zoneFile.vmiRecordsMap)
-
 	if isUpdated {
 		zoneFile.updateContent()
 	}
-	return isUpdated, nil
-}
-
-type ARecord struct {
-	value string
-	ip    string
+	return isUpdated
 }
 
-func isIfaceListChanged(ifacesMap map[string]ARecord, interfaces []v1.VirtualMachineInstanceNetworkInterface) (isChanged bool) {
-	if len(ifacesMap) != len(interfaces) {
-		return true
+func isEqual(arr1, arr2 []string) bool {
```

Comment:
better please array1, array2
I believe there is DeepEqual or something that already checks for it, better to use it 
or some existing IsEqual for arrays

Reply:
done